Let's try InstantOn on Intel Mac (#2670)
OpenLiberty/blogs: posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc

---
layout: post

> title: Let's try Open Liberty InstantOn on Intel Mac

Reviewer: ```suggestion
title: Let's try Liberty InstantOn on Intel Mac
```

# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/takakiyo
author_github: https://github.com/takakiyo

> seo-title: Let's try Open Liberty InstantOn on Intel Mac - OpenLiberty.io

Reviewer: ```suggestion
seo-title: Let's try Liberty InstantOn on Intel Mac - OpenLiberty.io
```

> seo-description: Let's try Open Liberty InstantOn on Intel Mac

Reviewer: ```suggestion
seo-description: Try Liberty InstantOn on a developer machine like a Mac system and see how you can achieve fast startup for cloud-native Java applications without compromise.
```

> blog_description: "Let's try Open Liberty InstantOn on Intel Mac"

Reviewer: ```suggestion
blog_description: "Let's try Liberty InstantOn on Intel Mac"
```

//open-graph-image: https://openliberty.io/img/blog/instantonperf.png
//blog-available-in-languages: ['ja']
additional_authors: 
- name: Hiroko Takamiya
  github: https://github.com/una-tapa
  image: https://avatars0.githubusercontent.com/una-tapa
---
= Let's try Open Liberty InstantOn on Intel Mac
Takakiyo Tanaka <https://github.com/takakiyo>
:imagesdir: /
:url-prefix:
:url-about: /

== What is Liberty InstantOn ? 

> Liberty InstantOn was announced on September 29, 2022. It is a technology that launches a full-featured Java EE application server in hundreds of milliseconds.

Reviewer: ```suggestion
Liberty InstantOn was announced on September 29, 2022. It is a technology that launches a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta EE APIs in hundreds of milliseconds.
```

Reviewer 2: YK's rewording is better.  But the current plan for InstantOn support is to only support Jakarta WebProfile and MicroProfile.

Something is missing here with respect to InstantOn for the application.  We are not trying to position InstantOn as a fast way to start the runtime.  Rather it should be focused on fast startup of the cloud-native application itself.  I suggest:

"It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds."

Author: Line 26 will be updated with Tom's suggestion
"It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds."

When using Serverless technology in a container environment, the startup time is important. The response time includes the time it takes for the container process to start.  So the startup time directly affects the quality of service.

> Many efforts have been made to reduce the startup time for Java applications. For example, one of the efforts is to use the capabilities of GraalVM to transform the application into a native executable. However, GraalVM does not allow all Java features to be used, so it is not possible to convert widely used Java EE applications into executables.

Reviewer: ```suggestion
Many efforts have been made to reduce the startup time for Java applications. For example, one of the efforts is to use the capabilities of GraalVM to transform the application into a native executable. However, GraalVM does not allow all Java features to be used, so it is not possible to convert widely used Java EE or Jakarta EE applications into executables.
```

Liberty InstantOn takes a totally different approach to reduce the startup time. It takes advantage of the Linux CRIU (Checkpoint/Restore In Userspace) technology. CRIU is a set of functions for recording (Checkpoint) the information of the running process in a file and restoring (Restore) the process from the file. 

> Open Liberty/WebSphere Liberty with an OpenJ9 VM incorporates CRUI technology to get a Checkpoint when the application finishes initializing. By committing the container in this state and resuming it from there, it can be started in a lightning fast time.

Reviewer: ```suggestion
Open Liberty (and WebSphere Liberty) with an Eclipse OpenJ9 VM incorporates CRIU technology to get a Checkpoint when the application finishes initializing. By committing the container in this state and resuming it from there, it can be started in a lightning fast time.
```

Reviewer 2: I would not mention WebSphere Liberty for an Open Liberty blog.  Also, we don't have any InstantOn support for WebSphere Liberty yet.  InstantOn is only in beta at this time and WebSphere Liberty does not have betas.

Reviewer 3: I don't believe `C` needs to be capitalized in "Checkpoint"

Author: Will remove WebSphere Liberty,  and change Checkpoint to checkpoint.

== Let's try InstantOn on Intel Mac

> The InstantOn steps in the blog article link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[Liberty InstantOn startup for cloud native Java applications] are  based on the Red Hat Enterprise Linux. The same experience can be achieved on an Intel Mac with macOS Monterey (version 12.x). This article shares the steps to experience the exciting technology on Mac. 

Reviewer 2: This begs the question.  Can this work with M1 Rosetta? If we know it doesn't work then we should mention it.

Author: Currently trying to chat with the techsales team. leaving the conversation open.  Will go through the doc and remove all "WAS Liberty" mention.

Author: Adding "Currently, InstantOn feature is not available on M1 aarch64 architecture. "

> The InstantOn sample environment is published as a container image from IBM Cloud Registry. As the container execution environment, we can use podman introduced by Homebrew. On Mac, we needed to tweak the steps a little bit to see the InstantOn in action. 

Reviewer: ```suggestion
The InstantOn sample environment is published as a container image from IBM Cloud Registry. As the container execution environment, we can use podman, which one can install via Homebrew. On Mac, we needed to tweak the steps a little bit to see the InstantOn in action. 
```

=== Preparing the Podman

Since InstantOn uses the CAP_CHECKPOINT_RESTORE function of the Linux kernel, the InstantOn feature does not work properly if podman is started in rootless mode, Also, once the virtual environment (podman machine) has been started in rootless mode, it will fail even if it is run in rootful mode. The solution is to create a new dedicated virtual machine.

The followng steps create a podman virtual environment and start it in rootful mode. We will configure it to connect to a virtual machine by default.

[source]
----
$ podman machine init rootful
Extracting compressed file
Image resized.
Machine init complete
To start your machine run:

	podman machine start rootful

$ podman machine set --rootful rootful
$ podman machine start rootful
Starting machine "rootful"
Waiting for VM ...
Mounting volume... /Users/takakiyo:/Users/takakiyo
API forwarding listening on: /var/run/docker.sock
Docker API clients default to this address. You do not need to set DOCKER_HOST.

Machine "rootful" started successfully
$ podman system connection default rootful-root
----

> Following the version of the podman used for this article. 

Reviewer 2: ```suggestion
The Following version of podman used for this article. 
```

Author: Thanks for catching.  Just realized it is also missing "was".  
The Following version of podman **was** used for this article.  
After committing the suggestion I will remember to add it.

----

Even without InstantOn, the application starts in 6 to 7 seconds in my environment. Not bad at all. From the command prompt in another window, use the curl command to confirm that the application started successfully. 

[source]
----
$ curl http://localhost:9080/dev/system/properties   
----

If you press Ctrl+C in the window that started the container, the container in which Liberty is running stops.

[source]
----
^C[AUDIT   ] CWWKE0085I: The server defaultServer is stopping because the JVM is exiting.
[AUDIT   ] CWWKE1100I: Waiting for up to 30 seconds for the server to quiesce.
[AUDIT   ] CWWKT0017I: Web application removed (default_host): https://de537b960bc9:9443/dev/
[AUDIT   ] CWWKT0017I: Web application removed (default_host): https://de537b960bc9:9443/health/
[AUDIT   ] CWWKT0017I: Web application removed (default_host): https://de537b960bc9:9443/metrics/
[AUDIT   ] CWWKT0017I: Web application removed (default_host): https://de537b960bc9:9443/ibm/api/
[AUDIT   ] CWWKZ0009I: The application guide-getting-started has stopped successfully.
[AUDIT   ] CWWKE0036I: The server defaultServer stopped after 2 minutes, 32.806 seconds.  
----

> ==== Getting the Checkpoint

Reviewer 2: ```suggestion
==== Checkpoint the application
```

[AUDIT   ] CWWKG0093A: Processing configuration drop-ins resource: /opt/ol/wlp/usr/servers/defaultServer/configDropins/defaults/keystore.xml
[AUDIT   ] CWWKG0093A: Processing configuration drop-ins resource: /opt/ol/wlp/usr/servers/defaultServer/configDropins/defaults/open-default-port.xml
[AUDIT   ] CWWKZ0058I: Monitoring dropins for applications.
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://940fd476eccc:9080/ibm/api/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://940fd476eccc:9080/health/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://940fd476eccc:9080/metrics/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://940fd476eccc:9080/dev/
[AUDIT   ] CWWKZ0001I: Application guide-getting-started started in 1.340 seconds.
[AUDIT   ] CWWKC0451I: A server checkpoint was requested. When the checkpoint completes, the server stops.
/opt/ol/wlp/bin/server: line 946:   130 Killed                  "${JAVA_CMD}" "$@" >> "${CHECKPOINT_CONSOLE_LOG}" 2>&1 < /dev/null 
----

> Since the container was started without `--rm`` this time, the container environment after execution remains as it is. In this environment, files that record process information with checkpoint are saved.

Reviewer 2: ```suggestion
This starts the application container and proceeds to start the application. After the application has started the runtime will perform a checkpoint of the process. When the process state has been saved the container exits. Since the container was started without `--rm` this time, the stopped container remains available for inspection. This stopped container, named `getting-started-checkpoint-container`, contains the checkpoint process state information.
```

[source]
----
$ podman ps -a
CONTAINER ID  IMAGE                             COMMAND               CREATED        STATUS                    PORTS       NAMES
940fd476eccc  localhost/getting-started:latest  /opt/ol/wlp/bin/s...  4 minutes ago  Exited (0) 4 minutes ago              getting-started-checkpoint-container
----

> The following command commits and register it as a new image.

Reviewer 2: ```suggestion
The following command commits the container into a new image.  This new image, named `getting-started-instanton`, will contain the checkpoint process state for the application. This process state is then used to quickly start the application when the `getting-started-instanton` image is run as a container.
```

[source]
----
$ podman commit getting-started-checkpoint-container getting-started-instanton
a856d767b8c31718dfbc6e60f742675448086fb4421490b5bfde6d3392d2f879
$ podman images
REPOSITORY                           TAG             IMAGE ID      CREATED         SIZE
localhost/getting-started-instanton  latest          a856d767b8c3  7 seconds ago   990 MB
localhost/getting-started            latest          1049db82664e  31 minutes ago  890 MB
icr.io/appcafe/open-liberty          beta-instanton  7cd2ddfe28da  2 weeks ago     864 MB
----

== Starting Open Liberty using the InstantOn feature 

> Now it's time to execute InstantOn. When you start Liberty using this image, the server will start in a flash.

Reviewer 2: ```suggestion
Now it's time to see how fast the `getting-started-instanton` application can start. When you start the application container using this image, the application will start in a flash.
```

[source]
----

> % podman run -it --rm --privileged -p 9080:9080 getting-started-instanton

Reviewer 2: Have you tried this without `--privileged`.  What about:
```
podman run \
  --rm \
  --cap-add=CHECKPOINT_RESTORE \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_PTRACE \
  -p 9080:9080 \
  getting-started-instanton
```

Reviewer 3: Yes I had the same question as Tom, it would be good to have the "unprivileged restore" being mentioned here in some way (either as something you tried and it worked, OR as something that is possible on Linux (as discussed in the original Liberty InstantOn blog) but needs further investigation on Mac).

Author: Currently trying to chat with the techsales team. leaving the conversation open.

Author: I was able to try and restore worked. The content was updated with more restore without --privileged info. Spell corrected

---
layout: post
title: Let's try Liberty InstantOn on Intel Mac
# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/takakiyo
author_github: https://github.com/takakiyo
seo-title: Let's try Liberty InstantOn on Intel Mac - OpenLiberty.io

> seo-description: Try Liberty InstantOn on Intel Mac

Reviewer: ```suggestion
seo-description: How developers can try Liberty InstantOn that enables millisecond startup of cloud-native Java applications on Intel Mac
```

Reviewer: Cc @lauracowen

blog_description: "Let's try Liberty InstantOn on Intel Mac"
//open-graph-image: https://openliberty.io/img/blog/instantonperf.png
//blog-available-in-languages: ['ja']
additional_authors: 
- name: Hiroko Takamiya
  github: https://github.com/una-tapa
  image: https://avatars0.githubusercontent.com/una-tapa
---
= Let's try Open Liberty InstantOn on Intel Mac
Takakiyo Tanaka <https://github.com/takakiyo>
:imagesdir: /
:url-prefix:
:url-about: /

== What is Liberty InstantOn ? 

> Liberty InstantOn provides a full-featured cloud-native Java runtime that supports the MicroProfile and Jakarta WebProfile APIs and enables applications to start in milliseconds.

Reviewer 4: ```suggestion
Liberty InstantOn provides a full-featured cloud-native Java runtime that supports the MicroProfile and Jakarta WebProfile APIs and enables applications to start in milliseconds. InstantOn is currently available as a special beta release container image.
```

Author: This is also great.  The commit suggestion button is grayed out at this time. I will come back again.

Author: My update including Vijay's comment crossed with David's suggestion. I pushed his suggested change.  Is "beta" twice a bit repetitive? I will wait for review.

> Liberty InstantOn was announced on September 29, 2022. It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds.

Reviewer 4: ```suggestion
Liberty InstantOn provides a full-featured cloud-native Java runtime that supports the MicroProfile and Jakarta WebProfile APIs and enables applications to start in milliseconds.
```

Reviewer 4: "hundreds of milliseconds" seems odd but might be fine. I first read it as "hundredths of milliseconds" but I dont know what the actual figure is- what if we just say "milliseconds" ?

> Liberty InstantOn was announced as a beta feature on September 29, 2022. It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds. InstantOn is currently available as a special beta release container image.

Reviewer 4: ```suggestion
Liberty InstantOn provides a full-featured cloud-native Java runtime that enables applications to start in hundreds of milliseconds and supports MicroProfile and Jakarta WebProfile APIs. InstantOn is currently available as a special beta release container image.
```

> When using Serverless technology in a container environment, the startup time is important. The response time includes the time it takes for the container process to start.  So the startup time directly affects the quality of service.

Reviewer 4: maybe 
```suggestion
When using serverless technology in a container environment, the startup time is important. Faster startup times mean faster response times. So the startup time directly affects the quality of service.
```

Many efforts have been made to reduce the startup time for Java applications. For example, one of the efforts is to use the capabilities of GraalVM to transform the application into a native executable. However, GraalVM does not allow all Java features to be used, so it is not possible to convert widely used Java EE or Jakarta EE applications into executables.

Liberty InstantOn takes a totally different approach to reduce the startup time. It takes advantage of the Linux CRIU (Checkpoint/Restore In Userspace) technology. CRIU is a set of functions for recording (Checkpoint) the information of the running process in a file and restoring (Restore) the process from the file. 

> Open Liberty with an Eclipse OpenJ9 VM incorporates CRIU technology to get a checkpoint when the application finishes initializing. By committing the container in this state and resuming it from there, it can be started in a lightning fast time.

Reviewer 4: ```suggestion
Open Liberty with an Eclipse OpenJ9 VM incorporates CRIU technology to get a checkpoint when the application finishes initializing. By committing the container in this state and resuming it from there, it can start in lightning fast time.
```

== Let's try InstantOn on Intel Mac

> The InstantOn steps in the blog article link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[Liberty InstantOn startup for cloud native Java applications] are based on the Red Hat Enterprise Linux. The same experience can be achieved on an Intel Mac with macOS Monterey (version 12.x). This article shares the steps to experience the exciting technology on Mac. Currently, InstantOn feature is not available on M1 aarch64 architecture. 

Reviewer 4: ```suggestion
The InstantOn steps in the  link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[Liberty InstantOn startup for cloud native Java applications] blog post are based on Red Hat Enterprise Linux. You can achieve the same experience on an Intel Mac with macOS Monterey (version 12.x). This post shares the steps to experience the exciting technology on Mac. Currently, the InstantOn feature is not available on M1 aarch64 architecture. 
```

> The InstantOn sample environment is published as a container image from IBM Cloud Registry. As the container execution environment, we can use podman, which one can install via Homebrew by `brew install podman`. On Mac, we needed to tweak the steps a little bit to see the InstantOn in action.

Reviewer 4: ```suggestion
The InstantOn sample environment is published as a container image from IBM Cloud Registry. We can use Podman as the container execution environment, which you can install via Homebrew by  running the`brew install podman` command. On Mac, we needed to tweak the steps a little bit to see the InstantOn in action.
```

=== Preparing the Podman

> Since InstantOn uses the CAP_CHECKPOINT_RESTORE function of the Linux kernel, the InstantOn feature does not work properly if podman is started in rootless mode, Also, once the virtual environment (podman machine) has been started in rootless mode, it will fail even if it is run in rootful mode. The solution is to create a new dedicated virtual machine. It may be necessary to remove existing VM by `podman machine list` and `podman machine rm` commands. 

Reviewer 4: ```suggestion
Since InstantOn uses the CAP_CHECKPOINT_RESTORE function of the Linux kernel, the InstantOn feature does not work properly if Podman is started in rootless mode. Also, once the virtual environment (Podman machine) is started in rootless mode, it will fail even if it is run in rootful mode. The solution is to create a new dedicated virtual machine. It might be necessary to remove any existing VMs by running the  `podman machine list` and `podman machine rm` commands. 
```

> The followng steps create a podman virtual environment and start it in rootful mode. We will configure it to connect to a virtual machine by default.

Reviewer 4: ```suggestion
The followng steps create a Podman virtual environment and start it in rootful mode. We will configure it to connect to a virtual machine by default.
```

Server:       Podman Engine
Version:      4.3.0
API Version:  4.3.0
Go Version:   go1.18.7
Built:        Fri Oct 21 04:16:35 2022
OS/Arch:      linux/amd64
----

=== Cloning and building the test application

Let's clone the sample application from GitHub and build it.

[source]
----
$ git clone https://github.com/openliberty/guide-getting-started.git
Cloning into 'guide-getting-started'...
remote: Enumerating objects: 2747, done.
remote: Counting objects: 100% (537/537), done.
remote: Compressing objects: 100% (275/275), done.
remote: Total 2747 (delta 172), reused 447 (delta 116), pack-reused 2210
Receiving objects: 100% (2747/2747), 859.87 KiB | 2.74 MiB/s, done.
Resolving deltas: 100% (1097/1097), done.
$ cd guide-getting-started/finish
$ mvn package
[INFO] Scanning for projects...
（...）
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.240 s
[INFO] Finished at: 2022-10-16T21:09:55+09:00
[INFO] ------------------------------------------------------------------------

----

=== Create a container image

> To use the InstantOn image, rewrite the first line of the Dockerfile in the same directory with `beta-instanton`` tag. If you want to play around, look inside this image and do your experiment! 

Reviewer 4: since the examples use podman, should it be"Containerfile" instead of "Dockerfile" ? I'm not sure

Author: Dockerfile is what the OL Guide sample provides. Podman can handle Dockerfile. 

https://docs.podman.io/en/latest/markdown/podman-build.1.html
> podman build Builds an image using instructions from one or more Containerfiles or Dockerfiles and a specified build context directory.

Author: Actually I should point to the original InstantOn-beta blog. It use the same OL guide sample that uses Dockerfile.
https://openliberty.io/blog/2022/09/29/instant-on-beta.html#dockerfile

[AUDIT   ] CWWKG0093A: Processing configuration drop-ins resource: /opt/ol/wlp/usr/servers/defaultServer/configDropins/defaults/keystore.xml
[AUDIT   ] CWWKG0093A: Processing configuration drop-ins resource: /opt/ol/wlp/usr/servers/defaultServer/configDropins/defaults/open-default-port.xml
[AUDIT   ] CWWKZ0058I: Monitoring dropins for applications.
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://940fd476eccc:9080/ibm/api/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://940fd476eccc:9080/health/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://940fd476eccc:9080/metrics/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://940fd476eccc:9080/dev/
[AUDIT   ] CWWKZ0001I: Application guide-getting-started started in 1.340 seconds.
[AUDIT   ] CWWKC0451I: A server checkpoint was requested. When the checkpoint completes, the server stops.
/opt/ol/wlp/bin/server: line 946:   130 Killed                  "${JAVA_CMD}" "$@" >> "${CHECKPOINT_CONSOLE_LOG}" 2>&1 < /dev/null 
----

> This starts the application container and proceeds to start the application. After the application has started the runtime will perform a checkpoint of the process. When the process state has been saved the container exits. Since the container was started without `--rm` this time, the stopped container remains available for inspection. This stopped container, named `getting-started-checkpoint-container`, contains the checkpoint process state information.

Reviewer 4: ```suggestion
This starts the application container and proceeds to start the application. After the application starts, the runtime performs a checkpoint of the process. When the process state has been saved, the container exits. Since the container was started without `--rm` this time, the stopped container remains available for inspection. This stopped container, named `getting-started-checkpoint-container`, contains the checkpoint process state information.
```

[AUDIT   ] CWWKZ0001I: Application guide-getting-started started in 0.066 seconds.
[AUDIT   ] CWWKC0452I: The Liberty server process resumed operation from a checkpoint in 0.131 seconds.
[AUDIT   ] CWWKF0012I: The server installed the following features: [cdi-3.0, checkpoint-1.0, concurrent-2.0, distributedMap-1.0, jndi-1.0, json-1.0, jsonb-2.0, jsonp-2.0, monitor-1.0, mpConfig-3.0, mpHealth-4.0, mpMetrics-4.0, restfulWS-3.0, restfulWSClient-3.0, servlet-5.0, ssl-1.0, transportSecurity-1.0].
[AUDIT   ] CWWKF0011I: The defaultServer server is ready to run a smarter planet. The defaultServer server started in 0.167 seconds.
----

Liberty started in 0.1-0.2 seconds in my environment. If you access it with the curl command or browser, you can see that the application is running successfully. Stop Liberty with Ctrl+C. This image can be used as many times as you want. 

== Starting Open Liberty using the InstantOn feature without `--privileged`

Running fully privileged containers is not recommended. The best practice is to instead reduce the elevated privileges down to only what is required to run the container. 

> It is possible to run podman with unconfined --security-opt options below also worked on the Mac OS mentioned below. For more details of the parameters and the input file, refer to link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[the original InstantOn blog] .

Reviewer 4: ```suggestion
You can run `podman` with the following unconfined --security-opt options, which also worked on the Mac OS as mentioned after the example. For more details of the parameters and the input file, refer to link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[the original InstantOn blog].
```

> podman command with unconfined `--security-opt` options

Reviewer 4: ```suggestion
The following example shows the `podman` command with unconfined `--security-opt` options
```

```
$ podman run \
  --rm \
  --cap-add=CHECKPOINT_RESTORE \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_PTRACE \
  --security-opt seccomp=unconfined \
  --security-opt systempaths=unconfined \
  --security-opt apparmor=unconfined \
  -p 9080:9080 \
  getting-started-instanton

[AUDIT   ] CWWKZ0001I: Application guide-getting-started started in 0.101 seconds.
[AUDIT   ] CWWKC0452I: The Liberty server process resumed operation from a checkpoint in 0.229 seconds.
[AUDIT   ] CWWKF0012I: The server installed the following features: [cdi-3.0, checkpoint-1.0, concurrent-2.0, ... transportSecurity-1.0].
[AUDIT   ] CWWKF0011I: The defaultServer server is ready to run a smarter planet. The defaultServer server started in 0.283 seconds
```

> podman command with seccomp to only required system calls. 

Reviewer 4: ```suggestion
The following example shows the `podman`  command with `seccomp` to only required system calls. 
```

```
$ podman run \
  --rm \
  --cap-add=CHECKPOINT_RESTORE \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_PTRACE \
  --security-opt seccomp=/Users/htakamiy@us.ibm.com/InstantOn/guide-getting-started/finish/criuRequiredSysCalls.json \
  -v /proc/sys/kernel/ns_last_pid:/proc/sys/kernel/ns_last_pid \
  -p 9080:9080 \
  getting-started-instanton

[AUDIT   ] CWWKZ0001I: Application guide-getting-started started in 0.101 seconds.
[AUDIT   ] CWWKC0452I: The Liberty server process resumed operation from a checkpoint in 0.220 seconds.
[AUDIT   ] CWWKF0012I: The server installed the following features: [cdi-3.0, checkpoint-1.0, concurrent-2.0, distributedMap-1.0, jndi-1.0, json-1.0, jsonb-2.0, jsonp-2.0, monitor-1.0, mpConfig-3.0, mpHealth-4.0, mpMetrics-4.0, restfulWS-3.0, restfulWSClient-3.0, servlet-5.0, ssl-1.0, transportSecurity-1.0].
[AUDIT   ] CWWKF0011I: The defaultServer server is ready to run a smarter planet. The defaultServer server started in 0.262 seconds.
```

> The podman command may no longer need to mount `ns_last_pid` nor `seccomp` parameter at a newer OS version. The following CHECKPOINT_RESTORE command worked on macOS Monterey 12.6.  

Reviewer 4: ```suggestion
The podman command might not  need to mount `ns_last_pid` nor `seccomp` parameter at a newer OS version. The following CHECKPOINT_RESTORE command worked on macOS Monterey 12.6.  
```

```
podman run \
  --rm \
  --cap-add=CHECKPOINT_RESTORE \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_PTRACE \
  -p 9080:9080 \
  getting-started-instanton
```

> The above command ran successfully on the following MacOS version. 

Reviewer 4: ```suggestion
The previous command ran successfully on the following MacOS version. 
```
